### ISSUE TITLE
Can't process MP3 files with VAD or diarization

### ISSUE THREAD

**Issue by Barabazs** [m-bain/whisperX#41]
Hi @m-bain. Thank you so much for your amazing work!

I wanted to test your new VAD feature, but I get an error stating that:
```
Traceback (most recent call last):
  File "/usr/local/bin/whisperx", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.8/dist-packages/whisperx/transcribe.py", line 451, in cli
    result = transcribe_with_vad(model, audio_path, vad_pipeline, temperature=temperature, **args)
  File "/usr/local/lib/python3.8/dist-packages/whisperx/transcribe.py", line 310, in transcribe_with_vad
    vad_segments = vad_pipeline(audio)
  File "/usr/local/lib/python3.8/dist-packages/pyannote/audio/core/pipeline.py", line 238, in __call__
    return self.apply(file, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pyannote/audio/pipelines/voice_activity_detection.py", line 197, in apply
    segmentations: SlidingWindowFeature = self._segmentation(file)
  File "/usr/local/lib/python3.8/dist-packages/pyannote/audio/core/inference.py", line 328, in __call__
    waveform, sample_rate = self.model.audio(file)
  File "/usr/local/lib/python3.8/dist-packages/pyannote/audio/core/io.py", line 278, in __call__
    waveform, sample_rate = torchaudio.load(file["audio"])
  File "/usr/local/lib/python3.8/dist-packages/torchaudio/backend/soundfile_backend.py", line 205, in load
    with soundfile.SoundFile(filepath, "r") as file_:
  File "/usr/local/lib/python3.8/dist-packages/soundfile.py", line 629, in __init__
    self._file = self._open(file, mode_int, closefd)
  File "/usr/local/lib/python3.8/dist-packages/soundfile.py", line 1183, in _open
    _error_check(_snd.sf_error(file_ptr),
  File "/usr/local/lib/python3.8/dist-packages/soundfile.py", line 1357, in _error_check
    raise RuntimeError(prefix + _ffi.string(err_str).decode('utf-8', 'replace'))
RuntimeError: Error opening 'xxxx.mp3': File contains data in an unknown format.
```
I've tried with different (valid) mp3 files and each time it results in this error.

I'm using whisperx on Google Collab.

// EDIT: I get the same error when I try the diarization feature.

**Comment by m-bain** [comment]
hmm yeah at the moment VAD assumes .wav file format
`ffmpeg -i audio.mp3 -ar 16000 -ac 1 audio.wav`

**Comment by Barabazs** [comment]
Apparently there is an upstream issue here: https://github.com/bastibe/python-soundfile/issues/349

**Comment by Infinitay** [comment]
tl;dr try to update your soundfile package to 0.12 or higher (https://github.com/pyannote/pyannote-audio/blob/b56add2e2033060dca28be71563dab5ff0fd3e64/requirements.txt#L17) as it may be out of date as mine as

If anyone else is having issues with this, try to update your version of the soundfile package. I ended up switching to `pipx` to use whisperx after parallel processing changes (if you haven't used it before I don't recommend it, I had other issues of my own e.g. packages such as soundfile not showing up in `pip list`) But I digress, I ran `pipx inject whisperx soundfile==0.12.1` and it updated my soundfile version to 0.12.0 (pyannote-audio maximum allowed version of the package at the time is 12).
